cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -139,18 +140,23 @@ protected void startUp() throws Exception {
       // Create a long running transaction
       Transaction transaction = txClient.startLong();
 
+      Map<String, File> localizedFiles = localizeUserResources(contextConfig, clientContext,
+                                                               localizeResources, tempDir);
+
       // Create execution spark context
-      final ExecutionSparkContext executionContext = sparkContextFactory.createExecutionContext(transaction);
+      final ExecutionSparkContext executionContext =
+        sparkContextFactory.createExecutionContext(transaction, localizedFiles);
 
-      // Add addition resources that needs to be localized to executor containers
+      // Add additional resources that need to be localized to executor containers
       if (!contextConfig.isLocal()) {
         Configuration hConf = contextConfig.set(executionContext).getConfiguration();
         if (pluginJar != null) {
           hConf = new Configuration(hConf);
           hConf.set(Constants.Plugin.ARCHIVE, pluginJar.getName());
         }
+        // also serialize the local resource names, so they can be recreated later.
+        hConf.set(Constants.Localization.LOCAL_RESOURCES, new Gson().toJson(localizedFiles.keySet()));
```

Comment:
Fixed. Also made the constants in SparkContextProvider package-private since they are not required outside the spark package.